### PR TITLE
Knmstate v0.20.0

### PR DESCRIPTION
**Pull request by qinqon**
This PR bump kubernetes-nmstate so webhook are moved to a Deployment and we don't run webhook at a privileged pod anymore, apart from that the cert-manager has being upgraded now it works fine at openshift and self generated keys/certs.

```release-note
knmstate bumped to v0.20.0
```

### CODE REVIEW

File: pkg/network/nmstate.go [hunk 1]
```diff
@@ -32,3 +39,36 @@ func renderNMState(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string,
...
+func cleanUpNMState(conf *opv1alpha1.NetworkAddonsConfigSpec, ctx context.Context, client k8sclient.Client) []error {
+	if conf.NMState == nil {
+		return nil
+	}
+
+	// Get existing
+	existing := &unstructured.Unstructured{}
+	gvk := schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"}
+	existing.SetGroupVersionKind(gvk)
+	namespace := "cluster-network-addons"
```
Comment (phoracek) [inline pkg/network/nmstate.go:52]: This should be taken from the operand namespace variable.

https://github.com/kubevirt/cluster-network-addons-operator/blob/master/pkg/network/multus.go#L58

File: data/nmstate/001-rbac.yaml [hunk 2]
```diff
@@ -113,9 +123,16 @@ rules:
   - nodes
   - namespaces
   verbs:
-  - get
+  - '*'
```
Comment (phoracek) [inline data/nmstate/001-rbac.yaml:126]: We talked about this one on the kubernetes-nmstate PR. Do we need this change?

File: data/nmstate/003-operator.yaml [hunk 3]
```diff
@@ -39,78 +40,74 @@ spec:
...
-      - name: dbus-socket
-        hostPath:
-          path: /run/dbus/system_bus_socket
-          type: Socket
-      - name: nmstate-lock
-        hostPath:
-          path: /var/k8s_nmstate
+        - name: tls-key-pair
+          secret:
+            secretName: {{template "handlerPrefix" .}}nmstate-webhook
```
Comment (phoracek) [inline data/nmstate/003-operator.yaml:76]: do we have support for this templating on CNAO?

File: data/nmstate/001-rbac.yaml (hunk 2 above)
Comment (qinqon) [reply data/nmstate/001-rbac.yaml:126]: I changed for cluster role not for normal role, I prefer to keep here as we have at knmstate.

File: data/nmstate/003-operator.yaml (hunk 3 above)
Comment (qinqon) [reply data/nmstate/003-operator.yaml:76]: CNAO is already dealing with it at master.

**Comment by phoracek** [comment]
/lgtm
/approve
/hold

Since this is a big change. Would you please deploy CNAO (at least with `make cluster-up cluster-sync`) and test that you are able to create a basic policy?

**Comment by qinqon** [comment]
/hold cancel
we can deploy nncp

**Comment by qinqon** [comment]
/hold

I am going to disable webhook, there is things to figure out at CNAO related to caBundle and secret.

File: pkg/network/nmstate.go [hunk 4]
```diff
@@ -32,3 +40,36 @@ func renderNMState(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir string,
...
 }
+
+func cleanUpNMState(conf *opv1alpha1.NetworkAddonsConfigSpec, ctx context.Context, client k8sclient.Client) []error {
+	if conf.NMState == nil {
+		return nil
+	}
+
+	// Get existing
+	existing := &unstructured.Unstructured{}
+	gvk := schema.GroupVersionKind{Group: "apps", Version: "v1", Kind: "Deployment"}
```
Comment (phoracek) [inline pkg/network/nmstate.go:51]: It is a DaemonSet

**Pull request merged**